rpc/core/src/model/message.rs

```diff
@@ -693,45 +695,82 @@ pub struct GetMetricsRequest {
 pub struct ProcessMetrics {
     pub resident_set_size: u64,
     pub virtual_memory_size: u64,
-    pub core_num: u64,
-    pub cpu_usage: f64,
-    pub fd_num: u64,
+    pub core_num: u32,
```

Comment:
There are multiple changes here similar to this changing `u64` to `u32`. Why is this necessary?

Reply:
Why would one use u64 to represent fields that can have a maximum value of like 128? :). 

I would scale some fields down to u16, but I generally try to keep variables as u32 because chances are the compiler might need to align it anyways.

In general I find that these days developers mismanage data sizes and use wrong data types for values that are finite in size. In this specific case you always ship extra for bytes with no purpose. A byte becomes a kilobyte, then a gigabyte and before you know it the VPS metrics are showing an extra terabyte of data consumed. This is not right.

I've been actually toying with an idea of converting all metrics to f32, but decided not to do it in case the data needs to be precise for some type of analysis.  Do note that when displaying it in logs you actually loose most of it.

OTOH a single poll for metrics is ~200 bytes, which means it's 1kB every 5 seconds. If an app decides to present this to the user (which is reasonable thing to do as a part of the node/server health display) and such app has 1,000 users we are now consuming 12mB per minute...

Also, unlike gRPC that does transport data compression, Borsh sends data raw.

So to summarize, this is me making small efforts to reduce data consumption one byte at a time and remove what I consider as incorrect types on fields that are finite in size.

Reply:
What about changing the request struct from 4 bools to the single u8

Reply:
U mean bitfields?  It won't be as easy to use in regular RPC like gRPC.

Technically IMO this should be a Vec of enums, and I was thinking about adding that, I also was thinking to add a dictionary as an auxiliary payload as @michaelsutton was suggesting to use a dictionary (but there are also cases where that's not easy).

But I decided to freeze everything as-is as in January I want to have a discussion about versioning this.  

When the RPC was being designed I was trying to steer the team toward wrapping Request/Response into version enums as Borsh does straight binary ser, but they didn't see the value in that. However recent discussions on the traits for data storage have made me realize that we can refactor wRPC to use that method on RPC structs and provide struct versioning this way.

So for now I think this should be left as-is and we should look at doing some improvements down the road.